### PR TITLE
test: increase RTOS test coverage with shared E2E and TP suites

### LINKED ISSUE #99: test: increase RTOS test coverage in CI (FreeRTOS, ThreadX, Zephyr)
## Problem

The FreeRTOS and ThreadX Linux-port test suites each contain a single monolithic test file (`test_freertos_core.cpp`, `test_threadx_core.cpp`) with a hand-rolled `CHECK` macro. They cover PAL conformance (threading, mutex, condition variable, memory pool) and basic core functionality (Message, Endpoint, SessionManager, Serializer), but skip all higher-level modules.

Meanwhile, the host POSIX test suite has 15 test files covering: message, endpoint, session_manager, serialization, e2e, transport (TCP/UDP), SD, RPC, events, TP, platform_threading, and PAL mock tests for each RTOS.

### Current RTOS test gaps

| Module | Host tests | FreeRTOS linux-port | ThreadX linux-port | Zephyr (native_sim) |
|--------|-----------|--------------------|--------------------|---------------------|
| Message / Serialization | test_message, test_serialization | basic checks | basic checks | not wired |
| Transport (UDP/TCP) | test_udp_transport, test_tcp_transport | none | none | none |
| Service Discovery | test_sd | none | none | none |
| RPC | test_rpc | none | none | none |
| Events | test_events | none | none | none |
| SOME/IP-TP | test_tp | none | none | none |
| E2E | test_e2e | none | none | none |
| PAL threading | test_platform_threading | mutex/cv/thread only | mutex/cv/thread only | none |
| PAL memory | (implicit) | pool alloc/release | pool alloc/release | none |

### Phase 1: Shared test logic (platform-independent)

- Extract the platform-independent test logic from the host Google Test files into header-only `.inc` or shared source files
- Create a lightweight test harness that works inside RTOS task entry points (FreeRTOS `xTaskCreate`, ThreadX `tx_application_define`) without requiring Google Test
- Alternatively, evaluate whether Google Test can run inside the RTOS Linux ports (it uses `main()` → would need adapter)

### Phase 2: Expand RTOS Linux-port tests

- Add transport tests (UDP at minimum) running on FreeRTOS POSIX port and ThreadX Linux port
- Add serialization roundtrip tests beyond the current basic checks
- Add E2E protection tests (platform-independent logic, should work on any backend)
- Add SOME/IP-TP segmentation/reassembly tests

### Phase 3: Zephyr native_sim coverage

- Wire up existing `zephyr/tests/test_core` to CI with `native_sim` board
- Add transport and SD tests using Zephyr's native networking stack
- Add to the preset-validation or a dedicated Zephyr CI workflow

### Phase 4: CI integration

- Add RTOS test results to CI dashboards / test reports
- Consider running RTOS Linux-port tests in the Docker dev container (required for ThreadX on macOS CI runners since ThreadX linux port needs `<sys/sysinfo.h>`)
- Add coverage instrumentation to RTOS test builds

## Acceptance criteria

- [ ] FreeRTOS and ThreadX linux-port tests cover at least: serialization, E2E, TP (platform-independent modules)
- [ ] Transport (UDP) tests run on at least one RTOS linux port
- [ ] Zephyr `native_sim` tests run in CI
- [ ] All RTOS test results are reported in CI (pass/fail visible in PR checks)
- [ ] No regression in existing host POSIX test suite

## Dependencies

- #95 (cross-compilation build system — presets and toolchain files)
- #96 (library consolidation — may simplify test link targets)

### PR DESCRIPTION
**Pull request by vtz**
## Summary

- Extract duplicated test logic (message, endpoint, session manager, serializer) from FreeRTOS, ThreadX, and Zephyr test files into `tests/shared/*.inc` includes — following the existing `pal_conformance_tests.inc` pattern
- Add platform-independent **E2E protection** and **SOME/IP-TP segmentation/reassembly** test suites to all three RTOS backends (FreeRTOS linux-port, ThreadX linux-port, Zephyr native_sim)
- Update CI path filters so `tests/shared/**` changes trigger all backend workflows

### New RTOS test coverage

| Module | Before | After |
|--------|--------|-------|
| Message / Serialization | basic checks | shared suite with roundtrip, boundary values, string, uint64 |
| E2E (CRC, header, protect/validate) | none | 25+ checks per backend |
| SOME/IP-TP (segment, reassemble, alignment, errors) | none | 20+ checks per backend |
| Endpoint | basic checks | shared suite with accessor verification |
| Session Manager | basic checks | shared suite with non-zero ID validation |

### Files changed

- `tests/shared/test_{message,endpoint,session_manager,serializer,e2e,tp}_common.inc` — new shared test includes
- `tests/freertos/test_freertos_core.cpp` — use shared includes, add E2E + TP
- `tests/threadx/test_threadx_core.cpp` — same
- `zephyr/tests/test_core/src/main.cpp` — same
- `zephyr/tests/test_core/prj.conf` — enable `CONFIG_SOMEIP_TP=y`
- `tests/{freertos,threadx}/CMakeLists.txt` — add project-root include path, link e2e/tp
- `CMakeLists.txt` — update Renode test targets with same include/link changes
- `.github/workflows/ci.yml` — add `tests/shared/**` to shared path filter

## Test plan

- [x] Host POSIX test suite passes (15/15 tests, no regression)
- [ ] FreeRTOS linux-port runtime tests pass (CI)
- [ ] ThreadX linux-port runtime tests pass (CI)
- [ ] Zephyr native_sim test_core passes (CI)
- [ ] FreeRTOS Renode test passes (CI)
- [ ] ThreadX Renode test passes (CI)

Closes #99

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **New Features**
  * Added shared test framework for core functionality verification across platforms.
  * Enabled enhanced memory safety features including stack overflow detection.

* **Tests**
  * Refactored existing tests to use centralized test suites for improved maintainability.
  * Expanded test coverage for E2E and transport protocol functionality.

* **Chores**
  * Updated build configurations and CI workflows for improved consistency.
  * Enabled transport protocol support in configuration.

### CODE REVIEW

File: tests/shared/test_tp_common.inc [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+              "tp_single_message_type");
+
+        manager.shutdown();
+    }
+
+    // Multi-segment: large payload gets split
+    {
+        TpManager manager(config);
+        manager.initialize();
+
```
[the remark this answers is not in the document]
Comment (vtz) [reply tests/shared/test_tp_common.inc]: Fixed in latest commits.

File: tests/shared/test_e2e_common.inc [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+#include "e2e/e2e_profiles/standard_profile.h"
+#include "someip/message.h"
+#include "common/result.h"
+#include <vector>
+
+static void test_e2e() {
+    using namespace someip;
+    using namespace someip::e2e;
+
+    printf("\n--- E2E protection tests ---\n");
```
[the remark this answers is not in the document]
Comment (vtz) [reply tests/shared/test_e2e_common.inc]: Fixed in latest commits.
